checkfiles.py

```diff
@@ -594,6 +660,20 @@ def compare_flowcell_details(flowcell_details_1, flowcell_details_2):
     return False
 
 
+def get_read_name_details(job_id, errors, session, url):
+    query = job_id +'?datastore=database&frame=object&format=json'
+    try:
+        r = session.get(urljoin(url, query))
+    except requests.exceptions.RequestException as e:
+        errors['lookup_for_read_name_detaisl'] = ('Network error occured, while looking for '
+                                                  'file read_name details on the portal. {}').format(str(e))
+    else:
+        details = r.json().get('read_name_details')
+        if details:
+            return details
+    return False
```

Comment:
This should return details or None.  You can remove the return False altogether since no return implies return None.  

This function returns to the variable 'read_name_details'.  If 'read_name_details' is False or None, then 'if read_name_details' has the same outcome.

False, None, empty string, empty list, empty dict...will all evaluate to false in a conditional.  

Also It's good practice to have boolean functions that return True/False or functions the return the object you want or None.  Mixing booleans with other return types is generally not needed.

Reply:
fixed